test/lit.cfg

```diff
@@ -2134,6 +2134,16 @@ if platform.system() != 'Darwin' or swift_test_mode == 'optimize_none_with_impli
 #
 
 config.substitutions.append(('%target-clangxx', '%s -std=c++11' % config.target_clang))
+config.substitutions.append(('%cxx-all-targets\(([^)]+)\)',
```

Comment:
I think that moving this to the next commit is better.  Additionally, I'm not sure how I feel about this - this is a really really really complex way to do this.  We already push lit well beyond what it should be used as, and this is particularly horrendous.  Can we use a different approach for this?  I think that adding the subdirectory as a multiple test suites in a lit.local.cfg might be better.